lib/trino-parquet/src/main/java/io/trino/parquet/reader/FilteredRowRanges.java

```diff
@@ -13,14 +13,47 @@
  */
 package io.trino.parquet.reader;
 
+import com.google.common.annotations.VisibleForTesting;
 import org.apache.parquet.internal.filter2.columnindex.RowRanges;
 
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.util.Objects.requireNonNull;
 
-public record FilteredRowRanges(RowRanges parquetRowRanges, long rowCount)
+// PrimitiveColumnReader iterates over RowRanges value-by-value
+// FlatColumnReader iterates over range of values using FilteredRowRangesIterator
+public record FilteredRowRanges(RowRanges parquetRowRanges, List<RowRange> rowRanges, long rowCount)
```

Comment:
This abstraction is error prone. It allows for an instance where the `RowRanges` does not agree with the `List<RowRange>`.

Reply:
It looks like I'll have to revert this to a normal class, I don't see a way to hide canonical constructor with record